src/DataStructures/Tensor/Identity.hpp

```diff
@@ -0,0 +1,34 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines the function identity
+
+#pragma once
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+/// \ingroup TensorGroup
+/// \brief returns the Identity matrix
+
+template <typename DataType, size_t Dim>
+Tensor<std::decay_t<tt::remove_reference_wrapper_t<DataType>>,
+       tmpl::integral_list<std::int32_t, 2, 1>,
+       index_list<SpatialIndex<Dim, UpLo::Up, Frame::NoFrame>,
+                  SpatialIndex<Dim, UpLo::Lo, Frame::NoFrame>>>
+identity(const std::array<DataType, Dim>& used_for_size) {
```

Comment:
Since the array is only used for size why pass the array and not just the DataType? I'm guessing it's relevant for some future PR.